### PR TITLE
Default to using sha256 for payload verification

### LINKED ISSUE #16: Support X-Hub-Signature-256
We should prefer this header and probably fall back to the `X-Hub-Signature` header in the case that it's missing (in the case of maybe old Enterprise Server installations???)

I'm also a little conflicted about the current practice of requiring a secret to be set. I think there are probably legitimate use cases where a secret isn't needed and it makes some approaches to local dev and testing a bit more complicated. For now let's provide an interface to opt-out and give a more kind error message (right now I'm pretty sure we throw a relatively opaque `KeyError` 😞) in the case that the user has not opted out and the header is missing.

### PR DESCRIPTION
**Pull request by bradshjg**
* `X-Hub-Signature-256` header is now preferred (legacy header still supported)
* Support disabling verification via the GITHUBAPP_SECRET setting (set to false to disable)

Fixes #16 

Related to #14

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -2,11 +2,31 @@
...
\ No newline at end of file
+    return app
+
+@pytest.fixture
+def github_app(app):
+    github_app = GitHubApp(app)
+
+    @github_app.on('issues')
+    def test_issue():
+        return 'issue event'
```
Comment (Napo2k) [inline tests/conftest.py:22]: I am not sure - but the github api docs will clear this out for sure - if events that support actions, like the `issues` event, will ever generate payloads _without_ actions. So this bit of test code might never be hit. However, I think it's good to test it!

**Review by Napo2k — APPROVED** [review]
This makes me happy :D

File: tests/conftest.py (hunk 1 above)
Comment (bradshjg) [reply tests/conftest.py:22]: What this is trying to capture (and should be documented better) is that it's possible to create a function that responds to every action related to an event. So in this case using `@github_app.on('issues')` will make the function trigger regardless of which action is in the payload.

The only use case I've had for it so far is with check runs where was have a `check_run` handler that starts and updates a thread in a chat channel (so folks can be pulled in and given context quickly if things go awry) and additionally a `check_run.completed` handler that will DM the committer if checks fail.

We could've implemented all of that logic in the `check_run` handler, but it felt a little nicer to separate those as in one case we're trying to leave a bit of an audit trail in a place where it's easy to get multiple eyes quickly and in the other case we're trying to make sure someone knows the code they pushed had checks fail and can make a decision as to whether they need some help or if they happened to make a easy to fix error.

Comment (Napo2k) [reply tests/conftest.py:22]: Ah, reacting to events overall, and also to their actionbs, makes a lot of sense.

**Pull request merged**